### PR TITLE
Rename `sendTyping` to `sendTypingIndicator` and add deprecation notes

### PR DESCRIPTION
**Pull request by compulim**
> In draft mode because this PR requires merge of #1541.

## Background

`sendTypingIndicator` is a better name for sending user's typing activity.

We are modifying a few areas in our code: actions, props, reducers, sagas, selectors. And adding a deprecation notes to them.

We will complete the deprecation and remove the code on 2020 January.

### CODE REVIEW

File: packages/component/src/Composer.js [hunk 1]
```diff
@@ -261,8 +266,14 @@ class Composer extends React.Component {
     props.dispatch(setSendTimeout(props.sendTimeout || 20000));
   }
 
-  setSendTypingFromProps(props) {
-    props.dispatch(setSendTyping(!!props.sendTyping));
+  setSendTypingIndicatorFromProps(props) {
+    if (typeof props.sendTyping === 'undefined') {
+      props.dispatch(setSendTypingIndicator(!!props.sendTypingIndicator));
+    } else {
+      // TODO: [P3] Take this deprecation code out when releasing on or after 2020 January 13
```
Comment (corinagum) [inline packages/component/src/Composer.js]:
```suggestion
      // TODO: [P3] Take this deprecation code out when releasing on or after January 13 2020
```

File: packages/component/src/Composer.js [hunk 2]
```diff
@@ -261,8 +266,14 @@ class Composer extends React.Component {
...
+      console.warn('Web Chat: "sendTyping" has been renamed to "sendTypingIndicator". Please use "sendTypingIndicator" instead. This deprecation migration will be removed on or after 2020 January 13.');
```
Comment (corinagum) [inline packages/component/src/Composer.js]:
```suggestion
      console.warn('Web Chat: "sendTyping" has been renamed to "sendTypingIndicator". Please use "sendTypingIndicator" instead. This deprecation migration will be removed on or after January 13 2020.');
```

File: packages/core/src/index.js [hunk 3]
```diff
@@ -43,11 +44,14 @@ export {
...
+  setSendTypingIndicator,
   startDictate,
   startSpeakingActivity,
   stopDictate,
   stopSpeakingActivity,
   submitSendBox,
-  version
+  version,
+
+  // TODO: [P3] Take this deprecation code out when releasing on or after 2020 January 13
```
Comment (corinagum) [inline packages/core/src/index.js]:
```suggestion
  // TODO: [P3] Take this deprecation code out when releasing on or after January 13 2020 
```

File: packages/core/src/reducer.ts [hunk 4]
```diff
@@ -20,6 +20,9 @@ export default combineReducers({
   referenceGrammarID,
   sendBoxValue,
   sendTimeout,
-  sendTyping,
-  suggestedActions
+  sendTypingIndicator,
+  suggestedActions,
+
+  // TODO: [P3] Take this deprecation code out when releasing on or after 2020 January 13
```
Comment (corinagum) [inline packages/core/src/reducer.ts]:
```suggestion
  // TODO: [P3] Take this deprecation code out when releasing on or after January 13 2020
```

File: packages/core/src/reducers/sendTypingIndicator.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { SET_SEND_TYPING } from '../actions/setSendTyping';
+import { SET_SEND_TYPING_INDICATOR } from '../actions/setSendTypingIndicator';
+
+const DEFAULT_STATE = false;
+
+export default function (state = DEFAULT_STATE, { payload, type }) {
+  switch (type) {
+    // TODO: [P3] Take this deprecation code out when releasing on or after 2020 January 13
+    case SET_SEND_TYPING:
+      console.warn('Web Chat: "sendTyping" has been renamed to "sendTypingIndicator". Please use "sendTypingIndicator" instead. This deprecation migration will be removed on or after 2020 January 13.');
```
Comment (corinagum) [inline packages/core/src/reducers/sendTypingIndicator.js]:
```suggestion
      console.warn('Web Chat: "sendTyping" has been renamed to "sendTypingIndicator". Please use "sendTypingIndicator" instead. This deprecation migration will be removed on or after January 13 2020.');
```

File: packages/core/src/sagas/sendTypingIndicatorOnSetSendBoxSaga.js [hunk 6]
```diff
@@ -9,29 +9,37 @@ import {
...
 const SEND_INTERVAL = 3000;
 
-function takeSendTyping(value) {
-  return take(({ payload, type }) => type === SET_SEND_TYPING && !payload.sendTyping === !value);
+function takeSendTypingIndicator(value) {
+  return take(
+    ({ payload, type }) => (
+      (type === SET_SEND_TYPING_INDICATOR && !payload.sendTypingIndicator === !value)
+
+      // TODO: [P3] Take this deprecation code out when releasing on or after 2020 January 13
```
Comment (corinagum) [inline packages/core/src/sagas/sendTypingIndicatorOnSetSendBoxSaga.js]:
```suggestion
      // TODO: [P3] Take this deprecation code out when releasing on or after January 13 2020
```

File: samples/07.b.customization-send-typing-indicator/index.html [hunk 7]
```diff
@@ -44,9 +44,9 @@
         window.WebChat.renderWebChat({
           directLine: window.WebChat.createDirectLine({ token }),
 
-          // Web Chat will send typing activity, on behalf of the user, when "sendTyping" is set to true.
+          // Web Chat will send typing activity, on behalf of the user, when "sendTypingIndicator" is set to true.
```
Comment (corinagum) [inline samples/07.b.customization-send-typing-indicator/index.html]:
```suggestion
          // Web Chat will send a typing activity, on behalf of the user, when "sendTypingIndicator" is set to true.
```

File: samples/07.b.customization-send-typing-indicator/index.html [hunk 8]
```diff
@@ -44,9 +44,9 @@
...
           // The typing indicator will be send when the send box by any means, including speech.
```
Comment (corinagum) [inline samples/07.b.customization-send-typing-indicator/index.html]:
```suggestion
          // The typing indicator will be sent when the send box by any means, including speech.
```

File: packages/component/src/Composer.js [hunk 9]
```diff
@@ -158,13 +158,13 @@ function patchPropsForAvatarInitials({ botAvatarInitials, userAvatarInitials, ..
   if (botAvatarInitials) {
     styleOptions = { ...styleOptions, botAvatarInitials };
 
-    console.warn('Web Chat: "botAvatarInitials" is deprecated. Please use "styleOptions.botAvatarInitials" instead.');
+    console.warn('Web Chat: "botAvatarInitials" is deprecated. Please use "styleOptions.botAvatarInitials" instead. This feature will be removed on or after 2019 December 11.');
```
Comment (corinagum) [inline packages/component/src/Composer.js]:
```suggestion
    console.warn('Web Chat: "botAvatarInitials" is deprecated. Please use "styleOptions.botAvatarInitials" instead. "botAvatarInitials" will be removed on or after December 11 2019 .');
```

File: packages/component/src/Composer.js [hunk 10]
```diff
@@ -158,13 +158,13 @@ function patchPropsForAvatarInitials({ botAvatarInitials, userAvatarInitials, ..
...
   }
 
   if (userAvatarInitials) {
     styleOptions = { ...styleOptions, userAvatarInitials };
 
-    console.warn('Web Chat: "userAvatarInitials" is deprecated. Please use "styleOptions.userAvatarInitials" instead.');
+    console.warn('Web Chat: "userAvatarInitials" is deprecated. Please use "styleOptions.userAvatarInitials" instead. This feature will be removed on or after 2019 December 11.');
```
Comment (corinagum) [inline packages/component/src/Composer.js]:
```suggestion
    console.warn('Web Chat: "botAvatarInitials" is deprecated. Please use "styleOptions.botAvatarInitials" instead. "botAvatarInitials" will be removed on or after December 11 2019 .');
```

File: packages/component/src/Composer.js [hunk 11]
```diff
@@ -239,8 +239,13 @@ class Composer extends React.Component {
       this.setSendTimeoutFromProps(props);
     }
 
-    if (!prevProps.sendTyping !== !sendTyping) {
-      this.setSendTypingFromProps(props);
+    if (
+      !prevProps.sendTypingIndicator !== !sendTypingIndicator
+
+      // TODO: [P3] Take this deprecation code out when releasing on or after 2020 January 13
```
Comment (corinagum) [inline packages/component/src/Composer.js]:
```suggestion
      // TODO: [P3] Take this deprecation code out when releasing on or after January 13 2020
```

File: packages/core/src/actions/setSendTyping.js [hunk 12]
```diff
@@ -1,6 +1,8 @@
+// TODO: [P3] Take this deprecation code out when releasing on or after 2020 January 13
 const SET_SEND_TYPING = 'WEB_CHAT/SET_SEND_TYPING';
 
 export default function (value) {
+  // Deprecation notes added to reducer
```
Comment (corinagum) [inline packages/core/src/actions/setSendTyping.js]: Move this line to line 2?

**Comment by cwhitten** [comment]
@compulim let's address the coverage and get this merged

**Pull request merged**